tests/perfalert/test_alerts/test_secretary_tool.py

```diff
@@ -0,0 +1,92 @@
+from datetime import (datetime,
+                      timedelta)
+
+import pytest
+import simplejson as json
+from mock import (Mock,
+                  patch)
+
+from treeherder.perf.models import (BackfillRecord,
+                                    BackfillReport,
+                                    PerformanceSettings)
+from treeherder.perf.secretary_tool import (SecretaryTool,
+                                            default_serializer)
+
+
+@pytest.fixture
+def performance_settings(db):
+    settings = {
+            "limits": 500,
+            "last_reset_date": datetime.utcnow(),
+        }
+    return PerformanceSettings.objects.create(
+        name="perf_sheriff_bot",
+        settings=json.dumps(settings, default=default_serializer),
+    )
+
+
+@pytest.fixture
+def expired_performance_settings(db):
+    settings = {
+            "limits": 500,
+            "last_reset_date": datetime.utcnow() - timedelta(days=30),
+        }
+    return PerformanceSettings.objects.create(
+        name="perf_sheriff_bot",
+        settings=json.dumps(settings, default=default_serializer),
+    )
+
+
+@pytest.fixture
+def create_record():
+    def _create_record(alert):
+        report = BackfillReport.objects.create(summary=alert.summary)
+        return BackfillRecord.objects.create(alert=alert, report=report)
+
+    return _create_record
+
+
+def test_secretary_tool_updates_only_matured_reports(test_perf_alert, test_perf_alert_2, create_record):
+    # create new report with records
+    create_record(test_perf_alert)
+    # create mature report with records
+    date_past = datetime.utcnow() - timedelta(hours=5)
+    with patch('django.utils.timezone.now', Mock(return_value=date_past)):
```

Comment:
I see why we reached this path.
Still, I noticed there's [this solution](https://stackoverflow.com/questions/7499767/temporarily-disable-auto-now-auto-now-add) which doesn't involve this mocking.

@airimovici what do you think?

Reply:
I think that mocking is a more robust way to do this, as it assures that everything that is done in that context is uses that custom date (the report and the record) and we don't have to manually change them afterwards.